src/components/editgrid/EditGrid.js

```diff
@@ -209,6 +209,7 @@ export default class EditGridComponent extends NestedArrayComponent {
     }
 
     this.components = this.components || [];
+    this.prevHasAddButton = this.hasAddButton();
```

Comment:
Shouldn't this be moved to the NestedArrayComponent class now?  I am concerned that all nested array components except EditGrid will "redraw" on first pass of checkData now.

Reply:
I hope I got it right